west.yml

```diff
@@ -23,6 +23,8 @@ manifest:
       url-base: https://github.com/zephyrproject-rtos
     - name: babblesim
       url-base: https://github.com/BabbleSim
+    - name: juju
+      url-base: https://juju.nz/src/michaelh
```

Comment:
...or you could ditch the hal entirely of course

Reply:
> ...or you could ditch the hal entirely of course

PLEASE

Reply:
@fabiobaltieri Thank you!

I have now been able to reduce the `hal_wch` repository to two header files - all based on the excellent and well maintained work @ https://github.com/cnlohr/ch32v003fun.

```
[dhiru@zippy wch]$ pwd
zephyrproject/modules/hal/wch

[dhiru@zippy wch]$ find . | grep -v git
.
./ch32v003fun
./ch32v003fun/funconfig.h <-- This can also go away
./ch32v003fun/ch32v003fun.h <-- We just need this
./zephyr
./zephyr/module.yml
./CMakeLists.txt
```

Do these two files still need to be part of the external `hal_wch` repository? Or can they somehow be included in this repository itself?

Reply:
The `hal_wch` stuff is badly named I believe. There is no traditional HAL present as such.

Reply:
I'd say that file can happily live somewhere in `soc/`, `ite` and `nuvoton/npcx` have no hal and their helper stuff is in a `common` directory there, that should be alright. @gmarull?

Reply:
> Brilliant! I'd say that file can happily live somewhere in `soc/`, `ite` and `nuvoton/npcx` have no hal and their helper stuff is in a `common` directory there, that should be alright. @gmarull?

Yep, as soon as license is compatible.

Zephyr should offer a kind of `svd2rust`, so people that want to write native code can do so in a standardized way. Reading driver code is a bit depressing today.

Reply:
The `hal_wch` module is gone now. We only depend on the in-tree `soc/wch/ch32v00x/ch32v003fun.h` header file which is included in this PR.

Reply:
The license for `soc/wch/ch32v00x/ch32v003fun.h` file seems to be https://github.com/cnlohr/ch32v003fun/blob/master/LICENSE. This file also has a copyright notice near the end of the file.

Reply:
I think the situation is a bit ambiguous, the repo has that MIT license but the note on the file seems to not imply any specific license, just require that the file is only used with their own devices. It'll have to be tagged explicitly to be part of the repository (`SPDX-License-Identifier: MIT` (there's another few files that have it) but we probably need to clarify that first.

@cnlohr do you happen to have more details about this file?

Reply:
https://github.com/search?q=repo%3Aopenwch%2Fch32v003%20Apache&type=code could be relevant here. A bunch of files are explicitly licensed under `Apache-2.0`.

This GitHub accounts links itself to https://x.com/patrick_riscv (the CTO at WCH).

Reply:
This is unclear.  There are files that WCH has published that are clearly under an extra clause, and we have completely avoided them (like the WCH USB-PD code).  But, there are some #define's that have had that weird license at the top, as I note here:

https://github.com/cnlohr/ch32v003fun/blob/master/ch32v003fun/ch32v003fun.h#L6696-L6716

But, AFAIK, there is no code present within it that are Apache-2.0 licensed.

Reply:
Also, AFAIK, you would only run into worse licensing issues if you use other more WCH-centric dev envs, since all of our actual code (not interface) is original work, under MIT.

Reply:
@cnlohr Thank you! The current version of the PR directly uses https://github.com/cnlohr/ch32v003fun/blob/master/ch32v003fun/ch32v003fun.h file only and nothing else.

I missed seeing https://github.com/cnlohr/ch32v003fun/blob/master/ch32v003fun/ch32v003fun.h#L6696-L6716 earlier which says,

```
/*
 * This file contains the contents of various parts of the evt.
 * 
 * The collection of this file was generated by cnlohr, 2023-02-18
 * and AlexanderMandera, 2023-06-23
 *
 * Contents subject to below copyright where applicable by law. 
 *
 * (IANAL, BUT Because it is an interface, it is unlikely protected by copyright)
 *
 *********************************** (C) COPYRIGHT *******************************
 * File Name          : ------------------
 * Author             : WCH
 * Version            : V1.0.0
 * Date               : 2020/08/08
 * Description        : Library configuration file.
*********************************************************************************
* Copyright (c) 2021 Nanjing Qinheng Microelectronics Co., Ltd.
* Attention: This software (modified or not) and binary are used for 
* microcontroller manufactured by Nanjing Qinheng Microelectronics.
*******************************************************************************/
```

@openwch Hey - Would it be possible to remove this extraneous restriction? Thank you!